### DISCUSSION TITLE
How to flash the klipper firmware on my sv06?

### DISCUSSION THREAD

**Discussion by hyzteric** [bassamanator/everything-sovol-sv06#40] [Q&A]
Hi,
I've been trying for an hour to flash the klipper firmware.
I have tried 3 different SD cards 8Gb and 16Gb, formatted FAT32 with 4k blocks. I mean I format them before every try. Even though I'm not sure it's needed.
I can put the sovol firmware and flash it, no problem, but when I put the klipper firmware on the sd card, the screen stays blue. No logo, nothing. And it's not blank, it looks like it's all lit up.
Also, it does something because, if I turn the printer off and on, the blue screen comes back and the printer is unusable. I can't connect to it from klipper.
I have to flash the sovol firmware again if I want to be able to use the printer again.
I don't know what am I doing wrong?
I also keep changer the firmware name (the .bin file I mean) every time as I read somewhere that it won't flash the same name twice.
I'm stuck :-/
Thanks for any help.

**Comment by bassamanator** [comment]
The most popular sovol repo on github, find it here: https://github.com/bassamanator/Sovol-SV06-firmware

Read the docs, and follow the instructions to a T.

**Reply by WhiteZeroX** [reply]
I followed the instructions to the S. Send help. 😫

**Reply by hyzteric** [reply]
I followed the instructions of course... but the firmware.bin would not flash.
I found that the sv06 Plus firmware would work on the sv06 too => download the firmware from the first answer in this post : https://www.reddit.com/r/Sovol/comments/17lep54/sv06_plus_klipper_install/

**Reply by bassamanator** [reply]
Please take note of the warning in this [section](https://github.com/bassamanator/Sovol-SV06-firmware?tab=readme-ov-file#flash-firmware):
> ⚠️ Many users have reported having issues flashing Klipper using the Sovol microSD card.

**Reply by hyzteric** [reply]
Yes I know, and I said :

> I have tried 3 different SD cards 8Gb and 16Gb, formatted FAT32 with 4k blocks

The issue was not the card, it was the .bin file.
I think all the SV06 do not have the same chips, that's why if would fail for some and not for other with the same file..

**Reply by bassamanator** [reply]
@hyzteric This would mean that Sovol would have to supply two versions of the firmware for the same printer, which is not the case as you can see on the sovol3d.com. Moreover, search my repo and you'll see that those who encountered flashing issues, resolved those issues with a different microSD card.

I'm already at the 'bottom of this' but I guess I could go deeper, but I don't have one of those 'problem' machines. If you'd like to look into this further, compile the firmware via [these](https://github.com/bassamanator/Sovol-SV06-firmware/discussions/111) instructions and try to flash.

**Reply by bassamanator** [reply]
> I think all the SV06 do not have the same chips, that's why if would fail for some and not for other with the same file..

Also wanted to mention that I've heard similar from others, however, have found nothing conclusive just yet.
